extension/k8sleaderelector/extension.go

```diff
@@ -27,6 +27,13 @@ type LeaderElection interface {
 
 // SetCallBackFuncs set the functions that can be invoked when the leader wins or loss the election
 func (lee *leaderElectionExtension) SetCallBackFuncs(onStartLeading StartCallback, onStopLeading StopCallback) {
+	lee.mu.Lock()
+	defer lee.mu.Unlock()
+
+	if lee.onStartLeadingChan != nil {
+		lee.onStartLeadingChan <- onStartLeading
+	}
+
 	lee.onStartedLeading = append(lee.onStartedLeading, onStartLeading)
 	lee.onStoppedLeading = append(lee.onStoppedLeading, onStopLeading)
```

Comment:
We need to pick only one communication channel. I don't think these lists are needed. They must be always empty because extensions start before receivers